### ISSUE TITLE
Skia Failing to render

### ISSUE THREAD

**Issue by TheOriginalBytePlayer** [skia4delphi/skia4delphi#145]
I have an application that heavy relies upon a quite a number of skia elements to render the interface, and normally it looks fine as you can see in the attached image below.

However, while I don't know the exact conditions, in certain cases when if the application gets backgrounded and then is brought back to the front and activated, none of the skia elements display, which, as you might imagine, is a bit of a problem!  

When they are buttons and things with multiple states I seem to be able to get them back by clicking on them but I need some global  method of resetting the display so they all appear.

CORRECT INTERFACE:

![2022-08-11-14-48-38](https://user-images.githubusercontent.com/18058224/184248445-7e03df1e-dd28-4374-97ff-b61fffb8381a.png)

STATE WITH NON-RENDERING SKIA ELEMENTS FOLLOWING BACKGROUNDING AND RESTORING

https://user-images.githubusercontent.com/18058224/184250490-00d5138d-4048-4438-b1f3-082f57917274.mp4

**Comment by checkdigits** [comment]
@TheOriginalBytePlayer  This is not related to the bug report - but what app is that? That UI is AMAZING. I'd really like to feature it in a future Embarcadero presentation at some point. EXCELLENT work!

**Comment by TheOriginalBytePlayer** [comment]
@checkDigits It's the new version (still heavily in development) of our current product FrameForge Studio which won your first or second cool product award.  I probably should have said to keep it confidential, so please talk to me before you talk about it, as we're still quite a ways off, but you ain't seen NOTHING yet... :)

**Comment by TheOriginalBytePlayer** [comment]
Anyone have any idea about some kind of skia command I can use to reset the skia system?

**Comment by checkdigits** [comment]
Don't worry, I'd never mention it without checking with you first. It looks epic - keep up the great work! 👏😁

**Comment by TheOriginalBytePlayer** [comment]
@viniciusfbb Not to bug you, but this is a bit of a show stopper for us.  If there is some kind of global Skia Reset command we could run when the application gets focus or some other way of Forcing skia to repaint everything that would be a great work around!

**Comment by viniciusfbb** [comment]
@TheOriginalBytePlayer
Hi! How is your configuration to enable Skia? (usually in your .dpr)
Are you changing the GlobalUseSkiaRasterWhenAvailable?

**Comment by viniciusfbb** [comment]
On Windows we made 2 backends for Canvas based on Skia:
1. Raster (CPU)
2. OpenGL (GPU)

In the future we will implement other backends, but first we need to find out which one is having the problem. 

Enabling Skia raster backend (CPU) on Windows:
```delphi
   GlobalUseSkia := True;
   GlobalUseSkiaRasterWhenAvailable := True;
```

Enabling the Skia OpenGL backend on Windows:
```delphi
   GlobalUseSkia := True;
   GlobalUseSkiaRasterWhenAvailable := False;
 ```

**Comment by peardox** [comment]
@viniciusfbb My recent issue with Python as per https://github.com/Embarcadero/python4delphi/issues/32 may be related to the problem  @TheOriginalBytePlayer is having as my own app was behaving extremely similarly to his before the fix. 

I say here "related" as I'm sure he's probably not using Python but the SetExceptionMask that was causing my own issues was resolved from your notes on the matter.

@TheOriginalBytePlayer If you have the source for all the packages you're using grep out SetExceptionMask and see what turns up - in my case I simply saved and restored the mask rather than the provided method (Set + Set different)

**Comment by TheOriginalBytePlayer** [comment]
Okay, thank you… I’ll see what I can come up with…

**Comment by TheOriginalBytePlayer** [comment]
To be honest, I'm a little leary of upgrading, as in the past, I've had to spend nearly a day to get it working with the other libraries that I use that depend on it.  When  I get time I'll give it a go, but we're in a bit of a crunch right now.  

Thank you, though, for the best support I've ever gotten on an open source software and I'll try and repay you as soon as is feasible by getting you the information you need.

**Comment by viniciusfbb** [comment]
@TheOriginalBytePlayer I understand! But when you do, you can follow the following step by step to avoid problems:

1 - Go to the current Skia4Delphi folder, open the `Skia4Delphi\Packages\RAD Studio 11 Alexandria\Skia.Package.RTL.dproj` project;
2 - After the IDE opens, right-click on this project and "Uninstall". A confirmation message will appear saying that this "will uninstall the packages... that depend on this one". Click OK;
3 - Download the new version of Skia4Delphi and install it in the same folder as the old one (the installer will automatically uninstall the old version);
4 - Open your packages that use Skia (which were consequently uninstalled in step2), and compile again and install them again;

**Comment by TheOriginalBytePlayer** [comment]
I have updated to the latest Skia via GetIt... Do you know if that is up to date?  So far, I have not seen this issue repeat itself with the new version but I will keep you updated.

**Comment by viniciusfbb** [comment]
> I have updated to the latest Skia via GetIt... Do you know if that is up to date?

The latest version is 4.1.1, you can check it in your GetIt window. Generally, Embarcadero keeps GetIt updated from the last two versions of RAD Studio, in this case, Sydney and Alexandria.

> So far, I have not seen this issue repeat itself with the new version but I will keep you updated.

Excellent! Please, if you have any news, let us know here, otherwise I will close the issue in the coming weeks.

**Comment by viniciusfbb** [comment]
Fixed in our new release [v5.0.0](https://github.com/skia4delphi/skia4delphi/releases)
